models/lfs_lock.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	api "code.gitea.io/sdk/gitea"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// LFSLock represents a git lfs lfock of repository.
+type LFSLock struct {
+	ID          int64       `xorm:"pk autoincr"`
+	RepoID      int64       `xorm:"INDEX"`
+	Repo        *Repository `xorm:"-"`
+	OwnerID     int64       `xorm:"INDEX"`
+	Owner       *User       `xorm:"-"`
+	Path        string
+	Created     time.Time `xorm:"-"`
+	CreatedUnix int64     `xorm:"INDEX"`
+}
+
+// BeforeInsert is invoked from XORM before inserting an object of this type.
+func (l *LFSLock) BeforeInsert() {
+	if l.CreatedUnix == 0 {
+		l.CreatedUnix = time.Now().Unix()
+	}
+	l.OwnerID = l.Owner.ID
+}
+
+// AfterLoad is invoked from XORM after setting the values of all fields of this object.
+func (l *LFSLock) AfterLoad() {
+	l.Created = time.Unix(l.CreatedUnix, 0).Local()
+	l.Owner, _ = GetUserByID(l.OwnerID)
+}
+
+func (l *LFSLock) loadAttributes(e Engine) error {
+	var err error
+	if l.Repo == nil {
+		l.Repo, err = GetRepositoryByID(l.RepoID)
+		if err != nil {
+			return err
+		}
+	}
+	if l.Owner == nil {
+		l.Owner, err = GetUserByID(l.OwnerID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// LoadAttributes load repo and publisher attributes for a lock
+func (l *LFSLock) LoadAttributes() error {
+	return l.loadAttributes(x)
+}
+
+// APIFormat convert a Release to lfs.LFSLock
+func (l *LFSLock) APIFormat() *api.LFSLock {
+	//TODO move to api
+	return &api.LFSLock{
+		ID:       strconv.FormatInt(l.ID, 10),
+		Path:     l.Path,
+		LockedAt: l.Created,
+		Owner: &api.LFSLockOwner{
+			Name: l.Owner.DisplayName(),
+		},
+	}
+}
+
+// IsLFSLockExist returns true if lock with given path already exists.
+func IsLFSLockExist(repoID int64, path string) (bool, error) {
+	return x.Get(&LFSLock{RepoID: repoID, Path: path}) //TODO Define if path should needed to be lower for windows compat ?
+}
+
+// CreateLFSLock creates a new lock.
+func CreateLFSLock(lock *LFSLock) (*LFSLock, error) {
+	err := CheckLFSAccessForRepo(lock.Owner, lock.RepoID, "create")
+	if err != nil {
+		return nil, err
+	}
+	isExist, err := IsLFSLockExist(lock.RepoID, lock.Path)
+	if err != nil {
+		return nil, err
+	} else if isExist {
+		l, err := GetLFSLock(lock.RepoID, lock.Path)
```

Comment:
`IsLFSLockExist` has been invoked on `GetLFSLock`.

Reply:
@lunny To be sure, I would better use directly GetLFSLock and check if `!IsErrLFSLockNotExist(err)` ?